src/ql/pass/map/qubits/map/detail/free_cycle.cc

```diff
@@ -127,105 +58,79 @@ utils::Bool FreeCycle::is_first_swap_earliest(
     utils::UInt fr0,
     utils::UInt fr1,
     utils::UInt sr0,
-    utils::UInt sr1
-) const {
+    utils::UInt sr1) const {
     if (options->reverse_swap_if_better) {
-        if (fcv[fr0] < fcv[fr1]) {
-            utils::UInt  tmp = fr1; fr1 = fr0; fr0 = tmp;
+        if (get_for_qubit(fr0) < get_for_qubit(fr1)) {
+            std::swap(fr0, fr1);
         }
-        if (fcv[sr0] < fcv[sr1]) {
-            utils::UInt  tmp = sr1; sr1 = sr0; sr0 = tmp;
+        
+        if (get_for_qubit(sr0) < get_for_qubit(sr1)) {
+            std::swap(sr0, sr1);
         }
     }
-    utils::UInt start_cycle_first_swap = utils::max(fcv[fr0] - 1, fcv[fr1]);
-    utils::UInt start_cycle_second_swap = utils::max(fcv[sr0] - 1, fcv[sr1]);
 
-    QL_DOUT("... fcv[" << fr0 << "]=" << fcv[fr0] << " fcv[" << fr1 << "]=" << fcv[fr1] << " start=" << start_cycle_first_swap << " fcv[" << sr0 << "]=" << fcv[sr0] << " fcv[" << sr1 << "]=" << fcv[sr1] << " start=" << start_cycle_second_swap << " is_first_swap_earliest=" << (start_cycle_first_swap < start_cycle_second_swap));
+    auto start_cycle_first_swap = utils::max(get_for_qubit(fr0) - 1, get_for_qubit(fr1));
+    auto start_cycle_second_swap = utils::max(get_for_qubit(sr0) - 1, get_for_qubit(sr1));
+
     return start_cycle_first_swap < start_cycle_second_swap;
 }
 
 /**
  * Returns what the start cycle would be when we would schedule the given
- * gate, ignoring resource constraints. gate operands are real qubit indices
- * and breg indices. Purely functional, doesn't affect state.
+ * gate, ignoring resource constraints. Purely functional, doesn't affect state.
  */
-utils::UInt FreeCycle::get_start_cycle_no_rc(const ir::compat::GateRef &g) const {
+utils::UInt FreeCycle::get_start_cycle(const ir::CustomInstructionRef &g) const {
     utils::UInt start_cycle = 1;
```

Comment:
I don't think it matters, since the mapper will only compare start cycles to choose the best one for routing. So whether it starts at 0 or 1 shouldn't matter?